cql/src/main/java/org/apache/cassandra/stargate/transport/internal/RequestToHeadersMapper.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.stargate.transport.internal;
+
+import java.net.InetSocketAddress;
+import java.util.*;
+
+public class RequestToHeadersMapper {
+  public static final String TENANT_ID_HEADER_NAME = "tenant_id";
+
+  @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+  public static Map<String, String> toHeaders(
+      Optional<InetSocketAddress> sourceAddressWithEncodedTenantId) {
+    Map<String, String> tenantIdHeaders = new HashMap<>();
+    sourceAddressWithEncodedTenantId
+        .map(tenantIdAddress -> UUID.nameUUIDFromBytes(tenantIdAddress.getAddress().getAddress()))
```

Comment:
I'm not sure this conversion is correct.

The proxy protocol [handler](https://github.com/stargate/stargate/blob/b04b68424634ca50c1f1fcae4b9aa966f6827c7c/cql/src/main/java/org/apache/cassandra/stargate/transport/internal/HAProxyProtocolDetectingDecoder.java#L73) creates the address as `new InetSocketAddress(msg.sourceAddress(), msg.sourcePort())`, so I believe the tenant ID is the host _name_ portion of this address. The binary address may be different depending on how load balancing is set up.

I'd assume we ought to use UUID.fromString(tenantIsAddress.getHostName())`

@tjake WDYT?